src/canonmn_int.cpp

```diff
@@ -2588,8 +2595,8 @@ std::ostream& printCsLensTypeByMetadata(std::ostream& os, const Value& value, co
     auto aperMaxTele = std::stof(base_match[4].str()) * tc;
     auto aperMaxShort = base_match[3].length() > 0 ? std::stof(base_match[3].str()) * tc : aperMaxTele;
 
-    if (flMin != exifFlMin || flMax != exifFlMax || exifAperMax < (aperMaxShort - .1 * tc) ||
-        exifAperMax > (aperMaxTele + .1 * tc)) {
+    if (flMin != exifFlMin || flMax != exifFlMax || exifAperMin < (aperMaxShort - .1 * tc) ||
+        exifAperMin < (aperMaxTele + .1 * tc)) {
```

Comment:
I'm not sure that I understand this change to use `exifAperMin` :thinking: 

From the lens description we get min & max focal length (24mm, 105mm)  and we get maximum aperture at the short and tele end. (F4, F7.1) 

Now if a lens is a possible match the maximum reported aperture should be between the `aperMaxShort` and `aperMaxTele`. 

I'm not sure how  `exifAperMin` which in this case is F40, would help in filtering out lenses?

Reply:
The problem of the Aperture value is that it's encoding and decoding aren't perfect, which is why we get these slightly incorrect values like 7.3 instead of 7.1.  

Can you double-check how exiftool handles this specific canon transformation?   

In general I'm ok with improving `canonEV` but we'd have to somehow make sure it doesn't have a negative impact on other exiv2 outputs.

Reply:
@hassec:
> Can you double-check how exiftool handles this specific canon transformation?

I forgot to mention that with ExifTool, the user's test file also outputs `Exif.CanonCs.MaxAperture` as `7.3`. On the code in git, apart from our Sigma exception, the canonEV() function looks the same on ExifTool and Exiv2.

> In general I'm ok with improving canonEV but we'd have to somehow make sure it doesn't have a negative impact on other exiv2 outputs.

Agreed. When I ran the Exiv2 testing on my changes, only 1 problem was flagged up.
```
======================================================================
ERROR: Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv_test (test_regression_allfiles.TestAllFiles)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:/Users/postscript-dev/source_code/exiv2_postscript-dev/tests/regression_tests/test_regression_allfiles.py", line 169, in test_func
    BT.reportTest(os.path.basename(filename), out)
  File "C:/Users\postscript-dev\source_code\exiv2_postscript-dev\tests/bash_tests/utils.py", line 567, in reportTest
    raise RuntimeError('\n' + log.to_str())
RuntimeError: 
[ERROR] The output of the testcase mismatch the reference
[INFO] The output has been saved to file C:/Users/postscript-dev/source_code/exiv2_postscript-dev/test/tmp/Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv.out
[INFO] simply_diff:
C:/Users/postscript-dev/source_code/exiv2_postscript-dev/test/data/test_reference_files/Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv.out: 227 lines
C:/Users/postscript-dev/source_code/exiv2_postscript-dev/test/tmp/Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv.out: 227 lines
The first mismatch is in line 90:
< Exif.CanonSi.ShutterSpeedValue               Short       1  184  1/54 s
> Exif.CanonSi.ShutterSpeedValue               Short       1  184  1/50 s
```

Looking at the [test/data/test_reference_files/Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv.out](https://github.com/Exiv2/exiv2/blob/1ff09506e036ab868e7992f64c0a90713ef32ab2/test/data/test_reference_files/Sigma_50mm_F1.4_DG_HSM_A_for_EOS.exv.out) file, I can see that it has
```
Exif.Photo.ShutterSpeedValue                 SRational   1  368640/65536  1/49 s
```
Although we don't have the makernotes specifications, the new `Exif.CanonSi.ShutterSpeedValue` value of `1/50` is closer to the `Exif.Photo.ShutterSpeedValue` value of `1/49` which looks like an improvement.

Do you have something else that you would like me to test for?